File: src/main/java/com/unical/travelapp/backend/booking/dto/PartenzaOrganizzatoreDto.java
package com.unical.travelapp.backend.booking.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PartenzaOrganizzatoreDto {

    private Long disponibilitaId;
    private LocalDateTime dataInizio;
    private LocalDateTime dataFine;
    private Integer postiDisponibili;
    private long numeroPrenotazioni;
    private long partecipantiTotali;
}

File: travelApp_frontEnd/app/src/main/java/com/example/travelapp/data/remote/dto/PartenzaOrganizzatoreDto.kt
package com.example.travelapp.data.remote.dto

import com.example.travelapp.domain.model.PartenzaOrganizzatore
import com.example.travelapp.domain.model.PrenotatoPartenza

data class PartenzaOrganizzatoreDto(
    val disponibilitaId: Long,
    val dataInizio: String? = null,
    val dataFine: String? = null,
    val postiDisponibili: Int? = null,
    val numeroPrenotazioni: Long = 0,
    val partecipantiTotali: Long = 0
)

fun PartenzaOrganizzatoreDto.toDomain(): PartenzaOrganizzatore =
    PartenzaOrganizzatore(
        disponibilitaId = disponibilitaId,
        dataInizio = dataInizio,
        dataFine = dataFine,
        postiDisponibili = postiDisponibili,
        numeroPrenotazioni = numeroPrenotazioni,
        partecipantiTotali = partecipantiTotali
    )

/**
 * La stessa prenotazione letta dal lato di chi vende: qui interessa chi e' il viaggiatore,
 * non se il viaggio e' recensibile.
 */
fun PrenotazioneDto.toPrenotatoPartenza(): PrenotatoPartenza =
    PrenotatoPartenza(
        prenotazioneId = id,
        nome = nomeViaggiatore,
        cognome = cognomeViaggiatore,
        numeroPartecipanti = numeroPartecipanti,
        prezzoTotale = prezzoTotale,
        statoPrenotazione = statoPrenotazione,
        statoPagamento = statoPagamento,
        dataPrenotazione = dataPrenotazione
    )
